From 28aa326b7d0068c78359f8bce00bef60e6c59d36 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Sat, 31 Mar 2012 13:12:09 +0200
Subject: [PATCH] usb-redir: Reset device address and speed on disconnect

Without this disconnected devices look like the last redirected device
in the monitor in "info usb".

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 usb-redir.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/usb-redir.c b/usb-redir.c
index e0eb393..a424826 100644
--- a/usb-redir.c
+++ b/usb-redir.c
@@ -1159,6 +1159,8 @@ static void usbredir_device_disconnect(void *priv)
         QTAILQ_INIT(&dev->endpoint[i].bufpq);
     }
     dev->interface_info.interface_count = NO_INTERFACE_INFO;
+    dev->dev.addr = 0;
+    dev->dev.speed = 0;
 }
 
 static void usbredir_interface_info(void *priv,
